### PR TITLE
[SYCL] Fix error when size_t is used as enum base

### PR DESCRIPTION
**Pull request by elizabethandrews**
"std::size_t" was emitted in the forward declaration of enum in integration header. 
This resulted in compilation errors about use of undeclared identifier "std". To fix this,
builtin is resolved to integral type in forward declaration.

### CODE REVIEW

File: clang/test/CodeGenSYCL/kernelname-enum.cpp
Comment (elizabethandrews) [inline clang/test/CodeGenSYCL/kernelname-enum.cpp:136]: Oh wait. I accidentally did not save the CHECK lines for this tes

**Review by Fznamznon — APPROVED** [review]

**Comment by bader** [comment]
> @intel/llvm-gatekeepers PR is ready for review and merge

@elizabethandrews, just wanted to clarify. Technically, only code owner's approval is needed and we already have it. Do you want to get more reviews from @intel/llvm-gatekeepers team before the merge?

**Comment by elizabethandrews** [comment]
> @intel/llvm-gatekeepers PR is ready for review and merge

Sorry no. I meant it is ready for merge. I thought gatekeepers always review code

**Comment by bader** [comment]
> I thought gatekeepers always review code

I don't review every PR ready for merge. Unfortunately, I don't have time to review all patches - there are too many of them. In addition to that I don't have expertise in all components, so I rely on code owners to do code review.

**Pull request merged**
